### ISSUE TITLE
Volley Http Client

### ISSUE THREAD

**Issue by halilozercan** [halilozercan/PleerSDK#2]
I think an http client like Volley or OkHttp would be better than a Utility class. Let's check them.

**Comment by myozka** [comment]
Why not retrofit?

**Comment by halilozercan** [comment]
Retrofit is higher level of OkHttp client and better serves for individual API requests. We need to put a cache layer in SDK. Also, request query might be needed. Maybe Retrofit works too but I didn't expect that. Let's look into that too.

**Comment by myozka** [comment]
I looked at [http://instructure.github.io/blog/2013/12/09/volley-vs-retrofit/](http://instructure.github.io/blog/2013/12/09/volley-vs-retrofit/) for time cmp
[https://www.captechconsulting.com/blogs/volley-vs-retrofit---a-battle-royale](https://www.captechconsulting.com/blogs/volley-vs-retrofit---a-battle-royale) for overall 
and [http://www.truiton.com/2015/04/android-volley-vs-retrofit-better-approach/](http://www.truiton.com/2015/04/android-volley-vs-retrofit-better-approach/) for I don't even know why.
Retrofit does not have a cache but much simpler according to those links.

**Comment by halilozercan** [comment]
It seems retrofit is the better choice in any way. However, they changed the basic parts in 2.0 version and there are not much example code for it.  We can either go with safe choice that is 1.9.0 and I'm using this version in my other apps or we can try the new library. What do you think?

**Comment by myozka** [comment]
If it were 1.8 to 1.9 I'd say use the older one but it is from 1.x to 2.x

It's safe to assume that they changed a lot in this update and for a while, at least, they will be using the -currently new- method signatures and structures.
**TL;DR:** Use the bleeding-edge version

**Comment by halilozercan** [comment]
Ok then. This closes #2, I'm opening a new issue for Retrofit.
